apps/api/src/modules/payments/payments.service.ts

```diff
@@ -759,6 +759,20 @@ export default class PaymentsService {
               status: circlePayment.status,
             })
             status = circlePayment.status
+            // If the new status is a final failed status, release the pack
```

Comment:
If we are changing a payment from pending to failed, release the pack. The logic where we make create the payment in circle doesn't catch this because we check if the payment could not be created, but even if it fails the payment is created as pending and later updated to failed in this task.